### PR TITLE
add RwaToken and RwaConduit

### PR DESCRIPTION
**Pull request by livnev**

### CODE REVIEW

File: src/RwaUrn.sol [hunk 1]
```diff
@@ -57,6 +63,7 @@ contract RwaUrn {
         daiJoin.exit(fbo, wad);
     }
     function wipe(uint256 wad) external operator {
+        daiJoin.join(address(this), wad);
```
Comment (godsflaw) [inline src/RwaUrn.sol:66]: good catch

File: src/RwaToken.sol [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    // --- Math ---
+    function add(uint x, uint y) internal pure returns (uint z) {
+        require((z = x + y) >= x);
+    }
+    function sub(uint x, uint y) internal pure returns (uint z) {
+        require((z = x - y) <= x);
+    }
+
+    constructor() public {
+        balanceOf[msg.sender] = 1 ether;
```
Comment (godsflaw) [inline src/RwaToken.sol:25]: Adjust `totalSupply` here too.

**Review by godsflaw — CHANGES_REQUESTED** [review]
One change requested in the constructor.

File: src/RwaConduit.sol [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+interface GemLike {
+    function transfer(address,uint256) external returns (bool);
+    function balanceOf(address) external returns (uint256);
+}
+
+
+contract RwaConduit {
+    GemLike public gov;
+    GemLike public dai;
+    address public to;
```
Comment (hexonaut) [inline src/RwaConduit.sol:12]: These could all be immutable to save gas.

File: src/RwaConduit.sol [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    constructor(address _gov, address _dai, address _to) public {
+        gov = GemLike(_gov);
+        dai = GemLike(_dai);
+        to = _to;
+    }
+
+    function push() public {
```
Comment (hexonaut) [inline src/RwaConduit.sol:20]: This should maybe be `external` to save memory copying.

File: src/RwaToken.sol [hunk 5]
```diff
@@ -0,0 +1,50 @@
+pragma solidity 0.5.12;
+
+contract RwaToken {
+    // --- ERC20 Data ---
+    string  public constant name     = "RWA Example Token";
+    string  public constant symbol   = "RWA001";
+    uint8   public constant decimals = 18;
+    uint256 public totalSupply;
+
+    mapping (address => uint)                      public balanceOf;
```
Comment (hexonaut) [inline src/RwaToken.sol:10]: `uint` should be `uint256`. Similarly in the rest of the file.

File: src/RwaToken.sol [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    constructor() public {
+        balanceOf[msg.sender] = 1 ether;
+    }
+
+    // --- Token ---
+    function transfer(address dst, uint wad) external returns (bool) {
+        return transferFrom(msg.sender, dst, wad);
+    }
+    function transferFrom(address src, address dst, uint wad)
+        public returns (bool)
```
Comment (hexonaut) [inline src/RwaToken.sol:33]: This should be `external` to save memory copy gas.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Looks good structurally. Added some requests for cosmetic / gas-saving changes.

File: src/RwaConduit.sol (hunk 4 above)
Comment (livnev) [reply src/RwaConduit.sol:20]: I don't think this makes a difference unless you have reference types in calldata arguments.

File: src/RwaToken.sol (hunk 6 above)
Comment (livnev) [reply src/RwaToken.sol:33]: See above.

**Pull request closed without merging**